bytecomp/matching.ml

```diff
@@ -2881,29 +2884,66 @@ and do_compile_matching repr partial ctx arg pmh = match pmh with
         (divide_record lbl.lbl_all (normalize_pat pat))
         ctx_combine repr partial ctx pm
   | Tpat_constant cst ->
+      let names = None in
       compile_test
         (compile_match repr partial) partial
         divide_constant
-        (combine_constant pat.pat_loc arg cst partial)
+        (combine_constant names pat.pat_loc arg cst partial)
         ctx pm
-  | Tpat_construct (_, cstr, _) ->
+  | Tpat_construct (_, cstr, pats) ->
+      let names_from_type_variant cstrs =
+        let (consts, blocks) = List.fold_left
```

Comment:
as we discussed, let's move this to bucklescript, also, guard it against Clflags.bs_only?

Reply:
Suggestions on where it should go?

Reply:
moved.